elf/elf.go

```diff
@@ -674,6 +695,17 @@ func (b *Module) Load(parameters map[string]SectionParams) error {
 					fd:    int(progFd),
 				}
 			}
+
+			if params, ok := parameters[secName]; ok && params.CustomConfig != nil {
```

Comment:
Further above we load programs from `elf.SHT_REL` - can those not be pinned?

Reply:
Even if not in the scope for this PR, we should think about how maps can be pinned to a custom path as well.

Reply:
> Further above we load programs from `elf.SHT_REL` - can those not be pinned?

Yes, they can. Ideally, the code should be refactored: we have duplicate code for bpf programs in SHT_REL and those that don't use symbols (i.e. programs that don't use maps). This duplication comes from the original code on [goebpf](https://github.com/safchain/goebpf/blob/master/goebpf.go).

If it makes things easier, we could focus on pinning bpf maps only and not bpf programs. And do the program pinning in a future PR. In practice, we have use cases for map pinning, but I can't think of a use case for program pinning at this moment. Up to you.